doc/design/model_market.md

```diff
@@ -0,0 +1,142 @@
+# Model Market
+
+The model market is a web site where model developers can publish and share their models and analysts can find some useful models to finish the analysis work. The model market can be deployed anywhere like on the cloud or on-premise with some configurations.
+
+In the [model zoo design](model_zoo.md), we described how model developers develop, publish and share custom models on SQLFlow, and how analysts can make use of the shared model with SQLFlow.
+
+Here, we'll describe how to build the model market model developers and analysts.
+
+## Overview
+
+The model market is designed to:
+
+1. Manage user login and logout.
+1. View published images available for the current user.
+1. View published trained models available for the current user.
+1. Publish (or remove) model definition images.
+1. Publish (or remove) trained models.
+1. Share (or remove share) model definition images to other users.
+1. Share (or remove share) trained models to other users.
+
+<p align="center">
+<img src="figures/model_market_overview.png">
+</p>
+
+In order to support publishing and sharing models securely, the model market is able to communicate with some [SSO](https://en.wikipedia.org/wiki/Single_sign-on) service to authenticate users. Only users that are logged in can do operations on the model market.
+
+Then model definition images and trained model can have below accessibility types:
+
+1. Private: Visible to the current user and users shared to.
+2. Public: Readable by every user.
+
+**NOTE: model definition and trained models have their own accessibility settings, e.g. a user uploaded a public ResNET model definition, then train the ResNET using a private image dataset, the trained model is set to private and only share to a few people.**
+
+## Steps to View Model Definitions and Trained Models
+
+<p align="center">
+<img src="figures/list_model_defs.png">
+</p>
+
+<p align="center">
+<img src="figures/list_trained_model.png">
+</p>
+
+1. **Optional**: Login to model market. If the user is not logged in, he can only view public model definitions and trained models.
+1. Click at the "Model Definitions" tab to see the list of model definition Docker images and the model class names in each Docker image.
```

Comment:
Besides model class, should we also show the parameters of the model definition?
Such as the parameter `wide_embeddings` and `deep_embeddings` for the model `wide_and_deep_classifier(wide_embeddings, deep_embeddings)`